Tensile/Source/lib/include/Tensile/Serialization/Predicates.hpp

```diff
@@ -223,6 +223,24 @@ namespace Tensile
                 iot::enumCase(io, value, "gfx1101", AMDGPU::Processor::gfx1101);
                 iot::enumCase(io, value, "gfx1102", AMDGPU::Processor::gfx1102);
             }
+
+            static inline std::unordered_map<std::string, AMDGPU::Processor> lookup
+                = {{"gfx803", AMDGPU::Processor::gfx803},
+                   {"gfx900", AMDGPU::Processor::gfx900},
+                   {"gfx906", AMDGPU::Processor::gfx906},
+                   {"gfx908", AMDGPU::Processor::gfx908},
+                   {"gfx90a", AMDGPU::Processor::gfx90a},
+                   {"gfx1010", AMDGPU::Processor::gfx1010},
+                   {"gfx1011", AMDGPU::Processor::gfx1011},
+                   {"gfx1012", AMDGPU::Processor::gfx1012},
+                   {"gfx1030", AMDGPU::Processor::gfx1030},
+                   {"gfx1031", AMDGPU::Processor::gfx1031},
+                   {"gfx1032", AMDGPU::Processor::gfx1032},
+                   {"gfx1034", AMDGPU::Processor::gfx1034},
+                   {"gfx1035", AMDGPU::Processor::gfx1035},
+                   {"gfx1100", AMDGPU::Processor::gfx1100},
+                   {"gfx1101", AMDGPU::Processor::gfx1101},
+                   {"gfx1102", AMDGPU::Processor::gfx1102}};
```

Comment:
If `Predicates.hpp` is included in N translation units, then N copies of this hash table will have to be initialized upon loading the Tensile library.

Reply:
We did this sort of thing with a regex in the rocsolver benchmark suite and it was a surprising source of slowdown. Once it was indirectly included in a few dozen translation units, it reached something like 10% of the runtime of the program when we were testing small matrix sizes.